c++/src/kj/compat/http.c++

```diff
@@ -3378,8 +3378,9 @@ private:
   kj::ArrayPtr<const byte> sendParts[2];
 
   kj::Maybe<kj::Array<byte>> queuedPong;
-  // If a Ping is received while currentlySending is true, then queuedPong is set to the body of
-  // a pong message that should be sent once the current send is complete.
+  // queuedPong holds the body of the next pong to write, cleared when the pong is written.  If a
+  // more recent ping arrives before the pong is actually written, we can update this value to
+  // instead respond to the more recent ping.
```

Comment:
I don't know that this is true. If we're writing the current pong, we may have sent some, but not all, of its bytes down the socket. If we have a pong AAAAAAAA and we overwrite it with BBBBBBBB, we might end up sending AAAABBBB and confusing the other end. Plus. there could be confusion if the pongs are of different length.

Would it help to have a "sending" pong and an "on-deck" pong so the on-deck pong could be freely overwritten?

Reply:
I think in this case the current implementation is OK, since `writeQueuedPong()` moves the pong content from `queuedPong` to `payload` just prior to the write call, then attaches `payload` to the write promise.  So `queuedPong` is effectively the "on-deck" pong, while the promise and its attached payload are effectively the "sending" pong, and there shouldn't be a case where a client could receive fragmented content from multiple pongs.

But I agree the comment could be confusing.  Perhaps it would be less confusing with "...cleared at the time the next pong write is set up" rather than "...cleared when the pong is written"?  Wasn't quite sure of the best way to phrase it because we have multiple levels of "queuing" going on.